### ISSUE TITLE
Interface Import - Add VLAN import

### ISSUE THREAD

**Issue by N0odlez** [netbox-community/netbox#14171]
### NetBox version

v3.6.4

### Feature type

New functionality

### Proposed functionality

I would like to see the Interface import function support the changing of the vlans tagged and the default vlan. This will help bulk update saving time. It will also help with new people joining the Netbox community to be able to import their infrastructure quicker and easier.

### Use case

Currently there is no way to bulk import interfaces and assign the vlan at the same time

**Comment by jdavidson2021** [comment]
I agree that the assignment of VLANs to interfaces is the biggest usability issue that we have with NetBox.  We can import pretty much everything with CSV, except for the assignment of VLANs to an interface.  This is currently a very manual process using the UI to bulk edit the interfaces on a device in order to assign the VLANs to interfaces on a network device.  It is very tedious and error prone.

**Comment by mrclschstr** [comment]
Currently we use the Netbox API as a workaround, but a bulk import with the specification of VLAN IDs would simplify the process a lot.

**Comment by Sislik** [comment]
Bump for feature. Hopefully someone will pick it up since it would be really handy to finally have it :)

**Comment by goteamkor** [comment]
any possibility of this making it into 4.0.x or the 4.1 train?
Running into this now with a new implementation of netbox with about 800 VM's. We can import the interfaces with the mode, but not the untagged vlan, or the multiple tagged vlans.

**Comment by jdavidson2021** [comment]
Any plans to tackle this feature in the near future?

**Comment by proy-rbbn** [comment]
I am hoping since you can bulk-edit vlans it isn't too much work to allow bulk-import of the same thing.... could someone pick that up in 4.1?

**Comment by pheus** [comment]
Hi, I’d be interested in contributing this feature. I noticed it’s not marked “needs owner” - is it okay if I take it on?

Also, should support for QinQ be included as part of this?

**Comment by sepetuks** [comment]
Just stumbled on this limitation as well and wonder if this is in near plans to be implemented?

**Comment by BartZimmo** [comment]
@jeremystretch Any news if @pheus can help with this feature?
The ability to update the tagged VLANs on interfaces would be a big addition to bulk import/update.

**Comment by ifoughal** [comment]
I've stumbled on this as of today as well and would be happy to contribute if need be.

**Comment by jeremystretch** [comment]
@BartZimmo @ifoughal would either of you like to volunteer to own this issue and submit a PR?

**Comment by ifoughal** [comment]
Hi @jeremystretch , would be happy to contribute! I can start working on it next week.

**Comment by pheus** [comment]
Hi @ifoughal and maintainers,

Back in April 2025, when I first offered to contribute on this issue, I put together a working patch to add tagged/untagged VLAN fields to the Interface and VMInterface CSV imports (tests included). If no one else has started and @ifoughal is okay with it, I’m happy to open a PR to finish this off. If you’d prefer to take it, I’m glad to collaborate or share my branch so we can co‑author - whatever works best. Thanks!

**Comment by ifoughal** [comment]
> Hi [@ifoughal](https://github.com/ifoughal) and maintainers,
> 
> Back in April 2025, when I first offered to contribute on this issue, I put together a working patch to add tagged/untagged VLAN fields to the Interface and VMInterface CSV imports (tests included). If no one else has started and [@ifoughal](https://github.com/ifoughal) is okay with it, I’m happy to open a PR to finish this off. If you’d prefer to take it, I’m glad to collaborate or share my branch so we can co‑author - whatever works best. Thanks!

I'm ok with it, please do open the PR & thank you for your contribution!

**Comment by jeremystretch** [comment]
Thanks @pheus, I've assigned this to you.

**Comment by jdavidson2021** [comment]
I'm very glad to see this feature implemented.  Thanks to everyone that played a part.  This is a significant upgrade when it comes to usability and the assignment of VLANs to interfaces.
